flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -66,6 +69,15 @@ bool PrescanAndSemaDebugAction::BeginSourceFileAction() {
 }
 
 bool CodeGenAction::BeginSourceFileAction() {
+  llvmCtx_ = std::make_unique<llvm::LLVMContext>();
+  if (this->currentInput().kind().GetLanguage() == Language::LLVM_IR) {
+    // Parse the bitcode...
+    SMDiagnostic err;
+    llvmModule_ = parseIRFile(currentInput().file(), err, *llvmCtx_);
+
+    return (nullptr != llvmModule_);
```

Comment:
`return !llvmModule_`? I don't think we need the Yoda-style or explicitness? See line 506 and 535

Reply:
So, we discussed this [before](https://github.com/flang-compiler/f18-llvm-project/pull/1422#discussion_r796839625) and decided that this is a good compromise between being compact and making the intent clear. I hope that that's fine :)

Btw: This change was already reviewed, accepted and merged: https://github.com/flang-compiler/f18-llvm-project/pull/1434

Reply:
This should be consistent though and `!llvmModule_` is used elsewhere in this patch. Also, the Yoda style turns up 0 times in MLIR and 3 times in LLVM (and 1 comment), so it appears to be less than popular.

@Leporacanthicus Can you put together a phabricator to make this change in llvm-project? Thanks.

Reply:
Just to clarify, I'm not married to the "Yoda style" and don't mind changing it. I use it out of habit - it's muscle memory. I used to work on a project where it was quite pervasive and to me it feels natural. But I agree that we should use a style that's more common and familiar. I do prefer `return (llvmModule_ != nullptr);` over `return !llvmModule_;` though.

> @Leporacanthicus Can you put together a phabricator to make this change in llvm-project? Thanks.

This code is not available in llvm-project/main :) I can fix it myself. In fact, I intend to refactor the driver to use the MLIR style (unless there are strong objections) and can make such tweaks at the same time. But I would like to sync-up `fir-dev` and `main` first (I mean the driver patches).

Reply:
Well defined pointer-to-bool conversion is.  Use it!